fotoapparat/src/main/java/io/fotoapparat/hardware/v1/FocusUpdater.java

```diff
@@ -0,0 +1,64 @@
+package io.fotoapparat.hardware.v1;
+
+import android.hardware.Camera;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import io.fotoapparat.hardware.Capabilities;
+import io.fotoapparat.parameter.FocusMode;
+
+import static io.fotoapparat.hardware.v1.capabilities.FocusCapability.toCode;
+import static io.fotoapparat.hardware.v1.capabilities.FocusCapability.toFocusMode;
+
+public class FocusUpdater {
+
+    private static final List<FocusMode> ACCEPTABLE_FOR_MANUAL_FOCUS_MODES = Arrays.asList(
+            FocusMode.AUTO, FocusMode.MACRO, FocusMode.CONTINUOUS_FOCUS
+    );
+
+    private final CameraParametersDecorator parameters;
+    private final Capabilities capabilities;
+
+    public FocusUpdater(CameraParametersDecorator parameters,
+                        Capabilities capabilities) {
+        this.parameters = parameters;
+        this.capabilities = capabilities;
+    }
+
+    public CameraParametersDecorator getParameters() {
+        return parameters;
+    }
+
+    /**
+     * @return whether or not camera should autoFocus after this
+     */
+    public boolean setManualFocus(List<Camera.Area> focusingAreas) {
```

Comment:
Can you use a different term than “manual” focus for this feature? It appears to be using auto focus and setting custom focus areas. I would recommend reserving the term “manual” focus for a feature that manually sets the focal length.